src/JLD.jl

```diff
@@ -1173,7 +1172,7 @@ macro save(filename, vars...)
         # Save all variables in the current module
         writeexprs = Vector{Expr}(undef, 0)
         m = __module__
-        for vname in names(m)
+        for vname in keys(m)
```

Comment:
This one looks like it might be a mistake? `m isa Module`?

Reply:
Good catch. How was this working before....

Reply:
I'm guessing it might not be tested properly. The two tests `git grep` finds in `jldtests.jl` both give an explicit list of variables, and this branch is for an implicit all-capture it seems.